### ISSUE TITLE
Versioned Transaction and Address Lookup Table feature activation

### ISSUE THREAD

**Issue by jstarry** [solana-labs/solana#26391]
#### Problem
Support for versioned transactions and the new address lookup table feature hasn't been enabled on the mainnet-beta cluster yet. Enabling [this feature](https://github.com/solana-labs/solana/issues/24047) is critical for allowing users to read / write more accounts in a single atomic transaction.

#### Rollout Plan
- Core Implementation
    - [x] https://github.com/solana-labs/solana/pull/17103 
        - Amendment 1: https://github.com/solana-labs/solana/pull/21576
        - Amendment 2: https://github.com/solana-labs/solana/pull/21634
    - [x] https://github.com/solana-labs/solana/pull/18725
    - [x] https://github.com/solana-labs/solana/pull/19139
    - [x] https://github.com/solana-labs/solana/pull/22223 (disabled until `versioned_tx_message_enabled` feature is activated)
    - [x] https://github.com/solana-labs/solana/pull/22201
    - [x] https://github.com/solana-labs/solana/pull/21616 (disabled until `versioned_tx_message_enabled` feature is activated)
        - Amendment 1: https://github.com/solana-labs/solana/pull/22011
        - Amendment 2: https://github.com/solana-labs/solana/pull/22221
        - Amendment 3: https://github.com/solana-labs/solana/pull/23678
- Core Audit
    - [ ] Schedule audits
    - [ ] Publish audits
- Client Implementation
    - [x] Rust SDK support for versioned transactions
    - [x] Rust SDK support for address lookup table instructions and state
    - [ ] web3.js support
    - [ ] Explorer support
    - [ ] CLI commands for creating and updating lookup tables
- Guides
    - [ ] Wallet Guide 
    - [ ] Application Guide
    - [ ] Explorer Guide
- [x] Activate `versioned_tx_message_enabled` feature on Testnet
- [x] Activate `versioned_tx_message_enabled` feature on Devnet
- [ ] Activate `versioned_tx_message_enabled` feature on Mainnet-Beta

**Comment by staccDOTsol** [comment]
Wen mainnet-beta?

got here via https://github.com/staccDOTsol/address-lookup-table-multi-swap

**Comment by abrozzesi** [comment]
Any update on this being merged ?

**Comment by jstarry** [comment]
The changes are merged but guides are still in progress. We are targeting a launch at the end of September assuming that all parties have enough time to prepare.

**Comment by Broever101** [comment]
> Is there any demo of using address lookup table and version transaction ?

This might help a little
https://medium.com/@antematterio/how-to-use-address-lookup-tables-in-solana-e2fe42dcc101

**Comment by pindaroso** [comment]
@jstarry Just checking in. Is this on mainnet? Thanks!

**Comment by Arrowana** [comment]
Note, activation tomorrow but latest cli does not have #27123 in `solana-cli 1.14.3`

**Comment by boucheravi** [comment]
thanks for the update!

**Comment by staccDOTsol** [comment]
one

more

epoch

**Comment by exp0nge** [comment]
Have there been noticeable getBlock perf. hits with this or did a new index get added that needs to be added to get the same perf as prior to activation?

**Comment by bmeeder22** [comment]
Did this ever get activated?

**Comment by a26nine** [comment]
> Did this ever get activated?

Yes, it did. A few weeks ago...

**Comment by omarkilani** [comment]
Hi @jstarry,

Just wondering if I'm understanding this correctly, but if an ALT is deactivated and closed, it *seems* to mean that a message which references that ALT can no longer load any addresses belonging to that ALT?

**Comment by jstarry** [comment]
> Hi @jstarry,
> 
> Just wondering if I'm understanding this correctly, but if an ALT is deactivated and closed, it _seems_ to mean that a message which references that ALT can no longer load any addresses belonging to that ALT?

Yeah, that's correct.

**Comment by statikdev** [comment]
>> Hi @jstarry,
>>
>> Just wondering if I'm understanding this correctly, but if an ALT is deactivated and closed, it seems to mean that a message which references that ALT can no longer load any addresses belonging to that ALT?
> 
> Yeah, that's correct.

If an account referenced in the ALT is no longer accessible, that would cause history to be lost from running `getSignaturesForAddress` or trying to associate that account to the transaction, right? 

What is the reason for allowing closing of an ALT?

**Comment by CriesofCarrots** [comment]
No, there shouldn't be any history lost. The data supporting `getSignaturesForAddress` and other RPC endpoints gets stored in rocksdb when the transaction is processed, and duplicated to bigtable from there.

**Comment by omarkilani** [comment]
> No, there shouldn't be any history lost. The data supporting `getSignaturesForAddress` and other RPC endpoints gets stored in rocksdb when the transaction is processed, and duplicated to bigtable from there.

Hey @CriesofCarrots , does this mean `loadedAddresses` from `getTransaction` still returns the right data even when the ALT account is closed?

**Comment by CriesofCarrots** [comment]
> Hey @CriesofCarrots , does this mean `loadedAddresses` from `getTransaction` still returns the right data even when the ALT account is closed?

Yes, that's correct.

**Comment by omarkilani** [comment]
> > Hey @CriesofCarrots , does this mean `loadedAddresses` from `getTransaction` still returns the right data even when the ALT account is closed?
> 
> 
> 
> Yes, that's correct.

Awesome!

Thanks for your quick response!

**Comment by statikdev** [comment]
> No, there shouldn't be any history lost. The data supporting getSignaturesForAddress and other RPC endpoints gets stored in rocksdb when the transaction is processed, and duplicated to bigtable from there.

👍 .

When walking backwards through transactions, if the ALT is not accessible, what would be the best way to find a referenced account?

**Comment by Huzaifa181** [comment]
> > Is there any demo of using address lookup table and version transaction ?
> 
> This might help a little https://medium.com/@antematterio/how-to-use-address-lookup-tables-in-solana-e2fe42dcc101

Any example to use to create the version transaction using PDA ?

**Comment by a26nine** [comment]
> > > Is there any demo of using address lookup table and version transaction ?
> > 
> > 
> > This might help a little https://medium.com/@antematterio/how-to-use-address-lookup-tables-in-solana-e2fe42dcc101
> 
> Any example to use to create the version transaction using PDA ?

https://blog.a26nine.dev/address-lookup-tables-and-versioned-transactions-on-solana

**Comment by Huzaifa181** [comment]
How to send the versioned transaction using the address lookup table through the serum multisig program?
Need to create any transaction as versioned transaction through serum multisig as to compress the limit size and send then execute
